### PR TITLE
Test/mygallery component

### PR DESCRIPTION
**Pull request by capetillo**
## TEST: `MyGallery` integration test

This test tests the way the component handles the api calls and how it handles the responses. I remount the component in every test so that there's no cross contamination between tests. In other words, to reset the state to a blank slate.
Here are the tests that I'm running
- renders a loading indicator on mounted (not an api related test)
- fetches thumbnails for each session on mount 
- renders sessions with thumbnails after loading 
- renders sessions in the correct descending order 
- does not render sessions without thumbnails

### CODE REVIEW

File: src/components/Images/MyGallery.vue [hunk 1]
```diff
@@ -40,43 +48,58 @@ const getThumbnails = async (sessionId) => {
       if (data.results.length > 0) {
         thumbnailsMap.value[sessionId] = data.results.map(result => result.url)
       }
-      loading.value = false
+      checkIfLoadingComplete()
```
Comment (jnation3406) [inline src/components/Images/MyGallery.vue:51]: why was this needed?

File: src/tests/integration/components/myGallery.test.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  const useConfigurationStore = defineStore('configuration', {
+    state: () => ({
+      thumbnailArchiveUrl: 'http://mock-api.com/'
+    })
+  })
+
+  // Initialize the stores
+  const sessionsStore = useSessionsStore()
+  const userDataStore = useUserDataStore()
+  const configurationStore = useConfigurationStore()
```
Comment (jnation3406) [inline src/tests/integration/components/myGallery.test.js:46]: Once you test more components will this stuff above here migrate to a shared test class that can be used by all the components? It seems like it would be common to most things you need to test.

**Review by jnation3406 — APPROVED** [review]
Looks good! I feel like I understand component testing a bit more now.

File: src/components/Images/MyGallery.vue (hunk 1 above)
Comment (capetillo) [reply src/components/Images/MyGallery.vue:51]: I was testing something out cause my tests weren't passing but it was not related and forgot to update!

File: src/tests/integration/components/myGallery.test.js (hunk 2 above)
Comment (capetillo) [reply src/tests/integration/components/myGallery.test.js:46]: yeah, I'll do that

File: package.json [hunk 3]
```diff
@@ -15,9 +15,11 @@
     "@fortawesome/free-solid-svg-icons": "^6.5.2",
     "@fortawesome/pro-regular-svg-icons": "^6.6.0",
     "@fortawesome/vue-fontawesome": "^3.0.6",
+    "@vitejs/plugin-vue": "^5.1.2",
```
[the remark this answers is not in the document]
Comment (mgdaily) [reply package.json:18]: Agreed, probably should be

**Review by mgdaily — APPROVED** [review]
